lib/inertia_rails/renderer.rb

```diff
@@ -105,19 +101,19 @@ def page
       }
     end
 
-    def deep_transform_values(hash, &block)
-      return block.call(hash) unless hash.is_a? Hash
-
-      hash.transform_values {|value| deep_transform_values(value, &block)}
-    end
+    def deep_transform_props(props, parent_path = '', &block)
+      props.reduce({}) do |transformed_props, (key, prop)|
+        current_path = [parent_path, key].reject(&:empty?).join('.')
```

Comment:
we can simplify this by sticking with arrays:

```suggestion
        current_path = path + [key]
```